moment-recur.js

```diff
@@ -252,6 +268,10 @@
             if (rule.measure === 'weeksOfMonthByDay' && !this.hasRule('daysOfWeek')) {
                 throw Error("weeksOfMonthByDay must be combined with daysOfWeek");
             }
+            
+            if (rule.measure === 'wholeMonths' && (!this.hasRule('daysOfWeek') || !this.hasRule('weeksOfMonthByDay'))) {
+            	throw Error("wholeMonths must be combined with weeksOfMonthByDay and daysOfWeek");
+            }
```

Comment:
This and the stuff in line 252 needs to go into some kind of validator function.  This function itself is very overstuffed.